Use `Buffer` from `buffer`

sjchmiela opened the pull request:
Unfortunately I couldn't figure out a TS configuration that **wouldn't** include Node types. It seems to me that `@types/bn.js` reference them explicitly in https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5919cb6bae26f700ea7f66ff39f46ea4ad2e00e3/types/bn.js/index.d.ts#L8 and… they just get pulled in.

vpontis commented:
@sjchmiela could we use `pnpm`s [patch](https://pnpm.io/cli/patch) to remove that line? Maybe that would make it easier to maintain?

sjchmiela commented:
Good idea! I haven't fully noticed `pnpm` supports patches now, that was my main complaint about `pnpm` haha!

- removed `node` references from `bn.js` and `tape` types, added `Buffer` to `bn.js`
- configured TS to:
  - emit `esnext` code following `glow-js` example
  - exclude `test` from typechecking
  - not import any `@types/` automatically (lets us have both `@types/node` which make `test` work and not include them when compiling `src`)

vpontis commented on package.json:
```diff
@@ -42,12 +42,14 @@
   "dependencies": {
     "ansicolors": "^0.3.2",
     "bn.js": "^5.2.0",
+    "buffer": "6.0.3",
```
Let's make this `^` since we aren't going to have auto upgrades on in this package. It's nice to pin deps in our repos but we don't want to in shared packages.

vpontis commented on tsconfig.base.json:
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es2018",
+    "target": "esnext",
```
Let's do `es6`, we want to make sure `shared` packages are more portable than not. `glow-js` uses `es6` now.

sjchmiela replied:
Lol, haven't had `master` pulled in `glow-js` when I checked so I missed https://github.com/glow-xyz/glow-js/pull/63. Will see what I can do!

vpontis replied:
Yea, so I went to `es6` since there were issues with `esnext` and `es2020` when I tried to import that code into Solana Explorer which uses create-react-app (that has some old babel stuff).

sjchmiela replied:
I don't fully get why they (or we) would set `target` to `es5` (`es6`) and add `esnext` to `lib`… The way I understand it (and I may be wrong) is that:
- TypeScript will let us use new features like `Object.values`, `[].includes`, `BigInt` (thanks to `esnext` in `lib`)
- but will try to compile modern structures like classes to older style code.

Which is what one would do if they knew the environment that the code will be run on supports some modern features (hence `lib`). But if we suspect the environment supports `esnext`, why not compile to `esnext`? Haha

Changed configuration around as suggested! [`0053451` (#3)](https://github.com/glow-xyz/beet/pull/3/commits/0053451da859efde0f8de11573e2db5eba5c3c1f)

<img width="850" alt="Zrzut ekranu 2022-07-22 o 18 14 58" src="https://user-images.githubusercontent.com/1151041/180480947-8ecef17f-f0e2-449b-ab2c-acaaf59e9299.png">
